### PR TITLE
More System.Dynamic Tests

### PR DESCRIPTION
**Pull request by JonHanna**
Starting testing on `CallSite` and `CallSiteBinder`. Pretty thorough tests for `InvokeMemberBinding`.

### CODE REVIEW

**Comment by bartdesmet** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test please (I/O error)

File: src/System.Linq.Expressions/tests/Dynamic/InvokeMemberBindingTests.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            Assert.Equal(value.GetType(), d.GetType());
+        }
+
+        [Theory, MemberData(nameof(ObjectArguments))]
+        public void InvokeCaseInsensitiveFails(dynamic value)
+        {
+            Assert.Throws<RuntimeBinderException>(() => value.tostring());
+        }
+
+        // TODO: Use a case-insensitive binder so that the above actually works.
```
Comment (stephentoub) [inline src/System.Linq.Expressions/tests/Dynamic/InvokeMemberBindingTests.cs]: Is there an issue for the TODO?

Comment (JonHanna) [inline src/System.Linq.Expressions/tests/Dynamic/InvokeMemberBindingTests.cs]: Raised, and link added to comment.

File: src/System.Linq.Expressions/tests/Dynamic/InvokeMemberBindingTests.cs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+        [Fact]
+        public void GenericType()
+        {
+            dynamic d = new List<int>();
+            dynamic x = 32;
+            d.Add(x);
+            d.Add(x);
+            int tally = 0;
+            foreach (var item in d)
```
Comment (JonHanna) [inline src/System.Linq.Expressions/tests/Dynamic/InvokeMemberBindingTests.cs]: Picked.

**Pull request merged**